From b55b43930d33835590e542cddfaafca3304564dc Mon Sep 17 00:00:00 2001
From: zachary zhang <zhangzg@marvell.com>
Date: Tue, 13 Feb 2018 08:04:03 +0800
Subject: [PATCH 1311/1345] dts: a70x0: add new dts file to support comexpress
 board

commit  354b86b93ea5decfb4ad61b68952d3785b75af8f from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

In comexpress board, SMII phy address is 0x10 instead of 0x1. This board
inherits other settings from a7020 AMC board.
This patch uses 0x10 to replace 0x1 phy address value and adds
make rule for this board dtb in makefile.

Change-Id: I98933593939ae877d4bbebbfab6a072e7778ca91
Signed-off-by: zachary zhang <zhangzg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/52307
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    3 +
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 .../boot/dts/marvell/armada-7020-comexpress.dts    |   74 ++++++++++++++++++++
 3 files changed, 78 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-7020-comexpress.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index eb36c581..6a93ce8 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -29,6 +29,9 @@ Boards:
    - "marvell,armada7020-amc"
       when the board being used is the Armada 7020 AMC board
 
+   - "marvell,armada7020-comexpress"
+      when the board being used is the Armada 7020 COMEXPRESS board
+
    - "marvell,armada7040-db-A", "marvell,armada7040-db"
       when the board being used is the Armada 7040 Development board
 
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 8690026..a2eabea 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -25,6 +25,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-nand.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-all-pcie.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7020-amc.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-7020-comexpress.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-70x0-customer.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-pcac.dtb
 dtb-$(CONFIG_ARCH_MVEBU_PD) += armada-7040-pd.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-7020-comexpress.dts b/arch/arm64/boot/dts/marvell/armada-7020-comexpress.dts
new file mode 100644
index 0000000..539253f9
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-7020-comexpress.dts
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2018 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 7020 COMEXPRESS board
+ * This board file supports the comexress board configuration
+ */
+
+#include "armada-7020-amc.dts"
+
+/ {
+	model = "Marvell Armada-7020 COMEXPRESS board setup";
+	compatible = "marvell,armada7020-comexpress", "marvell,armada7020",
+			"marvell,armada-ap806";
+
+	cp0 {
+		config-space {
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				status = "okay";
+				cp0_phy0: ethernet-phy@1 {
+					reg = <0x10>;
+				};
+			};
+
+			gop {
+				cp0_emac2: mac2 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&cp0_phy0>;
+				};
+			};
+		};
+	};
+};
-- 
1.7.9.5

